src/to-car.ts

```diff
@@ -0,0 +1,144 @@
+import fs from 'fs'
+import { Readable, Writable } from 'stream'
+
+import { CarWriter } from '@ipld/car'
+
+import { CID } from 'multiformats'
+import all from 'it-all'
+import pipe from 'it-pipe'
+
+import { sha256 } from 'multiformats/hashes/sha2'
+
+import normalizeAddInput from 'ipfs-core-utils/src/files/normalise-input/index'
+
+import globSource from 'ipfs-utils/src/files/glob-source'
+import { WriterChannel } from '@ipld/car/lib/writer'
+
+// TODO: Needs module with types commited + typedefinitions fixed (different cids...)
+// import importer from 'ipfs-unixfs-importer'
+const { importer }: any = require('ipfs-unixfs-importer')
+
+// TODO: create types.ts
+interface UnixFSEntryish {
+  type: 'file' | 'directory' | 'object' | 'raw' | 'identity' | string,
+  name: string
+  cid: CID
+  path: string
+  content: () => AsyncIterable<Uint8Array>
+}
+
+export async function packFileToCar ({ input, output }: { input: string | Iterable<string> | AsyncIterable<string>, output?: string }) {
+  const unixFsFiles = await streamToUnixFs({
+    source: normalizeAddInput(globSource(input, {
+      recursive: true
+    }))
+  })
+
+  await toCar({
+    files: unixFsFiles,
+    writable: fs.createWriteStream(output || unixFsFiles[0].path)
+  })
+}
+
+export async function packFileToCar2 ({input, output}: {input: string, output: string}) {
```

Comment:
I got to this secondary solution, given that I am not entirely satisfied with `packFileToCar` yet. This is more similar to what `js-ipfs` currently does, uses the block service to store things on the BlockStore. However, this solution seems odd.

What do you think @olizilla ?

Reply:
The initial solution also has the problem of putting everything in memory. So, I think we can probably create two function for `toCar`. One which receives an async iterable of `UnixFsEntry` and the other similar to this `packFileToCar2`

Reply:
my quick reaction is `packFileToCar2` is what i had in mind... wrapping up a carWriter so it can be used as the blockService that the unixfs importer writes to directly.

Reply:
so, should we drop the concept of a `toCar` alike for `unixFs -> car`, or do you not see a need for it? If no need, I can do some clean up on `packFileToCar2` to become the only one